copycat/runtime/src/main/java/org/apache/kafka/copycat/runtime/Worker.java

```diff
@@ -106,6 +112,17 @@ public void stop() {
         long started = time.milliseconds();
         long limit = started + config.getLong(WorkerConfig.TASK_SHUTDOWN_GRACEFUL_TIMEOUT_MS_CONFIG);
 
+        for (Map.Entry<String, Connector> entry : connectors.entrySet()) {
+            Connector conn = entry.getValue();
+            log.warn("Shutting down connector {} uncleanly; herder should have shut down connectors before the" +
```

Comment:
It's considered unclean shutdown because the way we expect this class to be used is that something external (i.e. the Herder) will be managing the lifetime of the connectors. If we get to `stop()` here and have not already shut down connectors and tasks, that means something is not properly tracking their lifecycle. For example, this could mean that if we wanted to report status info for connectors or if we ended up calling this because we had some uncaught exception such that we had no choice but to kill the process, we did not properly report that info.

We can change the terminology to something other than "unclean", but the intent was that `Worker` should be a "dumb" container that isn't really responsible for stopping things on its own. We could also just skip the `conn.stop()` and `task.stop()` calls, opting only to log the fact that we still have running connectors and tasks; this behavior just seemed nicer.